### ISSUE TITLE
Add possibility to pack texture or not before publish.

### ISSUE THREAD

**Issue by jeremy-bepoix** [ynput/ayon-blender#13]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

Currently, if you publish an asset, the textures are packed into the blend file. If you are not using site-sync or remote is non sense because all textures can be found in absolute, plus it makes your blender file considerably heavier.
[See : ](https://github.com/ynput/OpenPype/pull/2085)

Another thing, if you use image sequences in texture nodes and they don't start with 00, this breaks the texture package and won't publish too.

### How would you imagine the implementation of the feature?

Add toogle on frontend : Pack used images in the blend files.
mokup : 
![image](https://github.com/user-attachments/assets/fc3642ba-4705-46e1-b26c-ca4a9e3d01ab)

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

### Additional context:

Also see discussion here: https://discord.com/channels/517362899170230292/1248298172816822424/1263781294421315655

**Comment by BigRoy** [comment]
> Another thing, if you use image sequences in texture nodes and they don't start with 00, this breaks the texture package and won't publish too.

Interesting bug! :) Do you have more background details to this? Is this a known blender bug or a known limitation? Why does it fail?

---

Do you think it's worth having the settings define the default but potentially allowing to have an artist-facing toggle in the publisher UI so that if just one particular publish needed it, they can enable it. Or is defining the value in studio settings sufficient for production usage?

**Comment by jeremy-bepoix** [comment]
> Like the code they link @jeremy-bepoix - does it log a warning or error? Or does it 'pack' but just fail when loading?
Yep, is known limitation This produces an error if the start frame does not begin with 00
see : [bpy.types.Image.pack()](https://docs.blender.org/api/current/bpy.types.Image.html#bpy.types.Image.pack)

It's not easy in the sense that if you start to put your finger in the resource pack, it's going to get a bit complicated because you can pack each blender entitites. Pack Linked Libraries, pack Data, Pack Resources ect...
Then you can also un-pack..

The pack should be seen as a kind of archive. This could be useful for sending to a client, or a freelancer who doesn't have all the studio's resources.

I think it's best not to pack anything in the first place and maybe force absolute path for all files instead.
see : [bpy.ops.file.make_paths_absolute()](https://docs.blender.org/api/current/bpy.ops.file.html#bpy.ops.file.make_paths_absolute)

**Comment by LiborBatek** [comment]
It is very impracticall in first place (to pack all resources into asset scene file everytime).

I would go with ```ayon-blender``` addon setting for this ( ```pack textures``` option) and also make it optional too (so it can be exposed to the artist if needed).

However as we have it now, there is not happening any movement of resources/ file path changes when publishing takes place (as in maya) so this will need to be incorporated too....otherwise there could possibly happen harm if someone alter any resources...speaking of properly version those resources now...

Im also not sure if ```blender``` also have ```image``` loader for images as textures/image planes ...comming from e.g. Substance Painter (now mentioning it mostly because of versioning of such resources/ textures)

We def need to manage resources as in ```maya``` host imho as we lack it completely now - packing texture resources was kind of workaround to me from the very beginning.

**Comment by alainxi** [comment]
> 
> Im also not sure if `blender` also have `image` loader for images as textures/image planes ...comming from e.g. Substance Painter (now mentioning it mostly because of versioning of such resources/ textures)
> 

ayon-blender now has a `image` loader :

https://github.com/ynput/ayon-blender/pull/66

**Comment by LiborBatek** [comment]
yeah, ```image``` products can be loaded into ```blender``` already which is cool.

We should stick to have just an option to not ```Pack textures``` in the addons settings and do not touch any ```filepaths``` if disabled...so the user can have any sort of pathing inside the file nodes.
